lib/util/calculation.dart

```diff
@@ -0,0 +1,281 @@
+
+// for the arc(sine cosine tangent) functions to switch the answer to degrees if need be
+  static Function arcTrigWrapper(num arcTrigFunction(x)) {
+    num _convertedArcTrigFunction(num x) {
+      num value = arcTrigFunction(x);
+      if (usingDegrees) {
+        value *= 180 / PI;
+      }
+      return value;
+    }
+    return _convertedArcTrigFunction;
+  }
+
+// for the sine cosine tangent functions to switch the answer to degrees if need be
+  static Function trigWrapper(num trigFunction(x)) {
+    num _convertedTrigFunction(num x) {
+      if (usingDegrees) {
+        x *= PI / 180;
+      }
+      return trigFunction(x);
+    }
+    return _convertedTrigFunction;
+  }
+}
+
+abstract class UnaryCalculation extends Calculation {
+  const UnaryCalculation(Precedence levelOfImportance)
+      : super(levelOfImportance);
+
+  num call(num x);
+}
+
+abstract class BinaryCalculation extends Calculation {
+  const BinaryCalculation(Precedence levelOfImportance)
+      : super(levelOfImportance);
+
+  num call(num x, num y);
+}
+
+class BasicCalculation {}
+
+const Map<String, Calculation> _calculations = const {
+  add: const AddCalculation(),
+  arccos: const ArcCosineCalculation(),
+  arcsin: const ArcSineCalculation(),
+  arctan: const ArcTangentCalculation(),
+
+  cosine: const CosineCalculation(),
+
+  divide: const DivideCalculation(),
+  divide_human: const DivideCalculation(),
+
+  eToPower: const BaseECalculation(),
+  exponent: const ExponentCalculation(),
+  exponent_symbol: const ExponentCalculation(),
+
+  factorial: const FactorialCalculation(),
+  factorial_symbol: const FactorialCalculation(),
+
+  logOfX: const LogCalculation(),
+
+  multiply: const MultiplyCalculation(),
+
+  naturalLog: const NaturalLogCalculation(),
+  nthRootSymbol: const NthRootCalculation(),
+
+  percent_symbol: const PercentCalculation(),
+
+  sine: const SineCalculation(),
+  subtract: const SubtractCalculation(),
+  sqrtSymbol: const SqrtCalculation(),
+  squared: const SquaredCalculation(),
+
+  tangent: const TangentCalculation(),
+  tenToPower: const BaseTenCalculation(),
+
+  zerosShort: const ZerosCalculation()
+};
+
+getCalculation(String key) => _calculations[key] ?? key;
+
+
+class AddCalculation extends BinaryCalculation
+    implements BasicCalculation {
+
+  const AddCalculation() : super(Precedence.i);
+
+  num call(num x, num y) => x + y;
+}
+
+class ArcCosineCalculation extends UnaryCalculation {
+
+  const ArcCosineCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.arcTrigWrapper(acos)(x);
+}
+
+class ArcSineCalculation extends UnaryCalculation {
+
+  const ArcSineCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.arcTrigWrapper(asin)(x);
+}
+
+class ArcTangentCalculation extends UnaryCalculation {
+
+  const ArcTangentCalculation() : super(Precedence.v);
+
+  num call(num x) => Calculation.arcTrigWrapper(atan)(x);
+}
+
+
+class BaseTenCalculation extends UnaryCalculation {
```

Comment:
Seems like these two calculations should inherit from some base Base calculation. (Heh, base base...)

Perhaps a BaseNCalculation type that arbitrarily accepts a base for its constructor, and these two subclasses since they're the most popular use of it. It looks like ExponentCalculator is almost this.